sdk/instrumentation/bodyattribute/bodyattribute.go

```diff
@@ -69,3 +73,24 @@ func SetEncodedBodyAttribute(attrName string, body []byte, truncated bool, span
 		span.SetAttribute(fmt.Sprintf("%s.truncated", attrName), true)
 	}
 }
+
+// Largely based on:
+// https://github.com/jmacd/opentelemetry-go/blob/e8973b75b230246545cdae072a548c83877cba09/sdk/trace/span.go#L358-L375
+func truncateUTF8Bytes(b []byte, maxBytes int) []byte {
+	// We subtract 4 as that is the largest possible byte size for single rune
+	startIndex := maxBytes - 4
+	if startIndex < 0 {
+		startIndex = 0
+	}
+
+	for idx := startIndex; idx < maxBytes; {
+		_, size := utf8.DecodeRune(b[idx:])
```

Comment:
what is our assumption here? max-4 can land us in the middle of previous run and when we try DecodeRune from that index, it may give runeError which we are ignoring. 
So, are we just trying to make sure that we correct the last rune if it got corrupted because of truncation.
I am fine with that, just trying to make sure I understand the behaviour well and we add comments around these assumptions.